File: channel-manager/frontend/src/main/java/org/onehippo/cms7/channelmanager/ExtStoreFuture.js
/**
 * Copyright 2011 Hippo
 *
 * Licensed under the Apache License, Version 2.0 (the  "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
(function() {
    "use strict";

    Ext.namespace('Hippo.ChannelManager');

    Hippo.ChannelManager.ExtStoreFuture = Ext.extend(Hippo.Future, {

        constructor: function(config) {
            Hippo.ChannelManager.ExtStoreFuture.superclass.constructor.call(this, function(success, fail) {
                config.store.on('load', function() {
                    success({
                        store: config.store
                    });
                }, this, {single: true});
                config.store.on('exception', function() {
                    fail();
                }, this, {single: true});
                config.store.load();
            }.createDelegate(this));
        }

    });

    Ext.reg('Hippo.ChannelManager.ExtStoreFuture', Hippo.ChannelManager.ExtStoreFuture);

}());

File: channel-manager/frontend/src/main/java/org/onehippo/cms7/channelmanager/ExtStoreFuture.java
/*
 *  Copyright 2011-2013 Hippo B.V. (http://www.onehippo.com)
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package org.onehippo.cms7.channelmanager;


import org.json.JSONException;
import org.json.JSONObject;
import org.wicketstuff.js.ext.ExtObservable;
import org.wicketstuff.js.ext.data.ExtStore;
import org.wicketstuff.js.ext.util.ExtClass;
import org.wicketstuff.js.ext.util.JSONIdentifier;

@ExtClass("Hippo.ChannelManager.ExtStoreFuture")
public class ExtStoreFuture<T> extends ExtObservable {

    public static final String EXT_STORE_FUTURE = "ExtStoreFuture.js";
    private static final long serialVersionUID = 1L;

    private ExtStore<T> store;

    public ExtStoreFuture(ExtStore<T> store) {
        this.store = store;
    }

    @Override
    protected JSONObject getProperties() throws JSONException {
        JSONObject properties = super.getProperties();
        properties.put("store", new JSONIdentifier(this.store.getJsObjectId()));
        return properties;
    }

    public ExtStore<T> getStore() {
        return this.store;
    }

}
